### PR TITLE
Fixes Readme

### PR DESCRIPTION
**Pull request by venkatvghub**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,44 @@
...
+- Remove vendor lock-in (rely on OSS practically)
+- Kubernetes native (At the moment, we don't have extensions for non K8s solutions). Our environment is kubernetes native
+- Hassle Free onboarding - minimal changes to application and development lifecycle
+- Cost Effective - We should eventually be able to bill developers and teams on usage patterns. 
+- Slightly Opinionated - This isn't a PaaS offering at the momemt. And hence can be extended and deployed on any native kubernetes installation
+
+### Features
+- [ ] Build, Test and Deploy applications from laptop directly into kubernetes using simple CLI tools
+- [ ] Ship code to remote container without tunneling: Sync files into container directly using File Sync(using [Devspace](https://github.com/loft-sh/devspace))
+- [ ] Ability to provide hot reloading of apps : sync directly into the containers without restart
```
Comment (srinidhis05) [inline README.md:27]: Link of compiledaemon . static vs dynamic explanation

File: README.md [hunk 2]
```diff
@@ -1,2 +1,44 @@
...
+- [ ] Support for extensible custom Helm Hooks that handles provisioning of AWS infrastructure components using [LocalStack](https://github.com/localstack/localstack)
```
Comment (srinidhis05) [inline README.md:28]: Helm hooks reference

File: README.md [hunk 3]
```diff
@@ -1,2 +1,44 @@
...
+- [ ] Declaratively define service and service dependencies using [Helmfile](https://github.com/roboll/helmfile)
+- [ ] Traffic routing to right upstream (Using traefik 2.0 with custom ingress route)
+- [ ] Ability to selectively route traffic to different upstream(using header propogation)
+- [ ] Ability to expose preview URL for all services
+
```
Comment (srinidhis05) [inline README.md:33]: Add the hooks that are implemented and link for our repo itself

File: README.md [hunk 4]
```diff
@@ -1,2 +1,44 @@
...
+- Cost Effective - We should eventually be able to bill developers and teams on usage patterns. 
+- Slightly Opinionated - This isn't a PaaS offering at the momemt. And hence can be extended and deployed on any native kubernetes installation
+
+### Features
+- [ ] Build, Test and Deploy applications from laptop directly into kubernetes using simple CLI tools
+- [ ] Ship code to remote container without tunneling: Sync files into container directly using File Sync(using [Devspace](https://github.com/loft-sh/devspace))
+- [ ] Ability to provide hot reloading of apps : sync directly into the containers without restart
+- [ ] Support for extensible custom Helm Hooks that handles provisioning of AWS infrastructure components using [LocalStack](https://github.com/localstack/localstack)
+- [ ] Declaratively define service and service dependencies using [Helmfile](https://github.com/roboll/helmfile)
+- [ ] Traffic routing to right upstream (Using traefik 2.0 with custom ingress route)
```
Comment (srinidhis05) [inline README.md:30]: Traefik 2.0 header based routing reference

File: README.md [hunk 5]
```diff
@@ -1,2 +1,44 @@
...
+- [ ] Ability to selectively route traffic to different upstream(using header propogation)
```
Comment (srinidhis05) [inline README.md:31]: Opentelemetry baggage propagation link

File: README.md [hunk 6]
```diff
@@ -1,2 +1,44 @@
...
+- [ ] Ability to expose preview URL for all services
+
+## Docs
+
```
Comment (srinidhis05) [inline README.md:35]: Links to the devstack documentation architecture

File: README.md [hunk 7]
```diff
@@ -1,2 +1,44 @@
...
+### Features
+- [ ] Build, Test and Deploy applications from laptop directly into kubernetes using simple CLI tools
+- [ ] Ship code to remote container without tunneling: Sync files into container directly using File Sync(using [Devspace](https://github.com/loft-sh/devspace))
+- [ ] Ability to provide hot reloading of apps : sync directly into the containers without restart
+- [ ] Support for extensible custom Helm Hooks that handles provisioning of AWS infrastructure components using [LocalStack](https://github.com/localstack/localstack)
+- [ ] Declaratively define service and service dependencies using [Helmfile](https://github.com/roboll/helmfile)
+- [ ] Traffic routing to right upstream (Using traefik 2.0 with custom ingress route)
+- [ ] Ability to selectively route traffic to different upstream(using header propogation)
+- [ ] Ability to expose preview URL for all services
```
Comment (srinidhis05) [inline README.md:32]: Provide an example of this ingressroute app

File: README.md (hunk 1 above)
Comment (venkatvghub) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (venkatvghub) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (venkatvghub) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (venkatvghub) [reply README.md]: Done

File: README.md (hunk 6 above)
Comment (venkatvghub) [reply README.md:40]: Done

File: Architecture.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+2 different devstack implementations, the solution (using helm hooks) can provision different urls called feat1.example.com and feat2.example.com
+- Service Routing: The service routing is configured using Traefik 2.0's [IngressRoute](https://doc.traefik.io/traefik/v2.0/providers/kubernetes-crd/). In essence, as 
+part of the feature deployment, the helm hooks configure a preview url alongside a feature label(e.g. `feature1`) in this case. So, if an application receives a header called `feature1` or uses an endpoint like feature1.example.com, 
+  the ingress route sends the traffic to the feature 1 deployment/service. Else, its routed to the `Base Application`'s endpoint(in this case example.com). Lets explain with the following table:
+  
+  | Request                                              | Target Service |
+  |------------------------------------------------------|----------------|
+  | curl http://feature1.example.com                     | Feature 1      |
+  | curl -H 'uber-ctx-key: feature1' http://example.com  | Feature 1      |
```
Comment (srinidhis05) [inline Architecture.md:38]: uberctx-dev-label

**Pull request merged**